genesyscloud/integration_facebook/genesyscloud_integration_facebook_proxy.go

```diff
@@ -0,0 +1,181 @@
+	return p.conversationsApi.PostConversationsMessagingIntegrationsFacebook(*integrationFacebook)
+}
+
+// getAllIntegrationFacebookFn is the implementation for retrieving all integration facebook in Genesys Cloud
+func getAllIntegrationFacebookFn(ctx context.Context, p *integrationFacebookProxy) (*[]platformclientv2.Facebookintegration, *platformclientv2.APIResponse, error) {
+	var allFacebookIntegrationRequests []platformclientv2.Facebookintegration
+	const pageSize = 100
+
+	facebookIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsFacebook(1, 100, "", "", "")
+	if err != nil {
+		return nil, resp, fmt.Errorf("Failed to get facebook integration request: %v %v", err, resp)
+	}
+	if facebookIntegrationRequests.Entities == nil || len(*facebookIntegrationRequests.Entities) == 0 {
+		return &allFacebookIntegrationRequests, resp, nil
+	}
+
+	allFacebookIntegrationRequests = append(allFacebookIntegrationRequests, *facebookIntegrationRequests.Entities...)
+
+	for pageNum := 2; pageNum <= *facebookIntegrationRequests.PageCount; pageNum++ {
+
+		facebookIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsFacebook(pageNum, pageSize, "", "", "")
+		if err != nil {
+			return nil, resp, fmt.Errorf("Failed to get facebook integration request: %v %v", err, resp)
+		}
+
+		if facebookIntegrationRequests.Entities == nil || len(*facebookIntegrationRequests.Entities) == 0 {
+			break
+		}
+
+		allFacebookIntegrationRequests = append(allFacebookIntegrationRequests, *facebookIntegrationRequests.Entities...)
+	}
+
+	for _, facebookReq := range allFacebookIntegrationRequests {
+		rc.SetCache(p.facebookCache, *facebookReq.Id, facebookReq)
+	}
+
+	return &allFacebookIntegrationRequests, resp, err
+}
+
+// getIntegrationFacebookIdByNameFn is an implementation of the function to get a Genesys Cloud integration facebook by name
+func getIntegrationFacebookIdByNameFn(ctx context.Context, p *integrationFacebookProxy, name string) (string, bool, *platformclientv2.APIResponse, error) {
+	facebookIntegrationRequests, resp, err := getAllIntegrationFacebookFn(ctx, p)
+	if err != nil {
+		return "", false, resp, err
+	}
+
+	if facebookIntegrationRequests == nil || len(*facebookIntegrationRequests) == 0 {
+		return "", true, resp, fmt.Errorf("No integration facebook found with name %s", name)
+	}
+
+	var facebookIntegration platformclientv2.Facebookintegration
+	for _, facebookIntegrationRequest := range *facebookIntegrationRequests {
+		if *facebookIntegrationRequest.Name == name {
+			log.Printf("Retrieved the integration facebook id %s by name %s", *facebookIntegrationRequest.Id, name)
+			facebookIntegration = facebookIntegrationRequest
+			return *facebookIntegration.Id, false, resp, nil
+		}
+	}
+
+	return "", true, resp, err
```

Comment:
Here you will be returning `nil` for the error when you should be return `fmt.Errorf("No integration facebook found...")` as above